### PR TITLE
First attempt to document linkpac/aggregatepac

### PR DESCRIPTION
**Pull request by tomschr**
Described linking and aggregating.

### CODE REVIEW

File: xml/obs_basic_workflow.xml [hunk 1]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+   <sect3 xml:id="sec.obs.basicworkflow.reuse.linking">
+    <title>Linking a Package</title>
+    <para>
+     Basically, a linked package is a clone of another package with some
+     additional modifications. Linking is used in the following situations:
+    </para>
+    <itemizedlist>
+     <listitem>
+      <para>
+       Some changes in the source code is needed, but it is impossible,
```
Comment (sknorr) [inline xml/obs_basic_workflow.xml]: No "some", no "is" & more grammar fixes necessary. Proposal: "The source code needs changes, but the source either cannot be changed or it is impractical or inconvenient to change the source."

File: xml/obs_basic_workflow.xml [hunk 2]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+       unpractical, or inconvenient to change it.
+      </para>
+     </listitem>
+     <listitem>
+      <para>
+       Separate the original source from own patches.
```
Comment (sknorr) [inline xml/obs_basic_workflow.xml]: "To separate..." (because you are giving a reason)

File: xml/obs_basic_workflow.xml [hunk 3]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+      </para>
+     </listitem>
+    </itemizedlist>
+    <para>
+     The general signature of the <command>linkpac</command> command is:
```
Comment (sknorr) [inline xml/obs_basic_workflow.xml]: signature->syntax

File: xml/obs_basic_workflow.xml [hunk 4]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+      >SOURCEPAC</replaceable> <replaceable
+       >DESTPRJ</replaceable></screen>
+    <para>
+     For example, to link the package <package>python-lxml</package> from
+     <uri>devel:language:python</uri> into your home project, use the
+     following command:
+    </para>
+    <screen>&prompt.user;&osccmd; linkpac devel:language:python python-lxml &obshome1;</screen>
+    <para>
+     In contrast to aggregating, this time the checkout contains all the
```
Comment (sknorr) [inline xml/obs_basic_workflow.xml]: remove "this time"

File: xml/obs_basic_workflow.xml [hunk 5]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+     files from the linked repository. To reduce it to a single file (like
+     in aggregating) you need to <quote>unexpand</quote> it in the working
```
Comment (sknorr) [inline xml/obs_basic_workflow.xml]: replace:
"like in" -> "like with"
"you need to" -> ", "

File: xml/obs_basic_workflow.xml [hunk 6]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
    <title>Reusing Packages in Your Project</title>
-   <para/>
-   <remark>TBD</remark>
+   <para>
+    To reuse existing packages in your package repository, &obsa; offers
+    two methods: <quote>aggregating</quote> and <quote>linking</quote>.
+   </para>
+
+   <remark>toms 2017-08-30: add a notes when NOT to use the two methods.
+   </remark>
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: True, I hope some experts can contribute.

File: xml/obs_basic_workflow.xml [hunk 7]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+
+   <sect3 xml:id="sec.obs.basicworkflow.reuse.aggregating">
+    <title>Aggregating a Package</title>
+    <para>
+     Basically, an aggregated package is a pointer to the build results
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Ok, removed.

File: xml/obs_basic_workflow.xml [hunk 8]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+     (binaries) of a package, copying the build results into another
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Ok, discussed with Stefan. Seems the word "aggregate" (see its [definition)](https://www.merriam-webster.com/dictionary/aggregate) is used in different way than OBS means it.
Leaving a comment for further discussion.

File: xml/obs_basic_workflow.xml [hunk 9]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+    <itemizedlist>
+     <listitem>
+      <para>
+       No changes in the source code are needed.
+      </para>
+     </listitem>
+     <listitem>
+      <para>
+       Packages are needed as a build requirement but are available
+       only in a different project.
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Ok, used your definition, thanks! 😉

File: xml/obs_basic_workflow.xml [hunk 10]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+      </para>
+     </listitem>
+     <listitem>
+      <para>
+       Build results (binaries) of the aggregated package are copied
+       into the target's package repository.
+      </para>
+     </listitem>
+     <listitem>
+      <para>Package rebuilds are avoided.</para>
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Ok, fixed.

File: xml/obs_basic_workflow.xml [hunk 11]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+     </listitem>
+    </itemizedlist>
+
+    <para>
+     The general signature of the <command>aggregatepac</command> command is:
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Ok, done.

File: xml/obs_basic_workflow.xml [hunk 12]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+       >DESTPRJ</replaceable></screen>
+    <para>
+     For example, to aggregate the package <package>python-lxml</package>
+     from <uri>devel:language:python</uri> into your home project,
+     use the following command:
+    </para>
+    <screen>&prompt.user;&osccmd; aggregatepac devel:language:python python-lxml &obshome1;</screen>
+    <para>
+     The next time you update your working directory, the <filename>python-lxml</filename>
+     directory is downloaded with the <filename>_aggregate</filename> file.
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: From an idiomatic point, your sentence is better. However, it's wrong. The files are not replaced, actually there aren't any (source) files, just the `_aggregate` file. So I used:

```
The next time you update your working directory, the <filename>python-lxml</filename>
contains the <filename>_aggregate</filename> file.
```

File: xml/obs_basic_workflow.xml [hunk 13]
```diff
@@ -616,8 +616,117 @@ Requires:         zool >= 1.5.6</screen>
...
+    </para>
+   </sect3>
+   <sect3 xml:id="sec.obs.basicworkflow.reuse.linking">
+    <title>Linking a Package</title>
+    <para>
+     Basically, a linked package is a clone of another package with some
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Ok, done!

File: xml/obs_basic_workflow.xml (hunk 1 above)
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Yeah, added that.

File: xml/obs_basic_workflow.xml (hunk 2 above)
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Right!

File: xml/obs_basic_workflow.xml (hunk 3 above)
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Done

File: xml/obs_basic_workflow.xml (hunk 4 above)
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Ok, done

File: xml/obs_basic_workflow.xml (hunk 5 above)
Comment (tomschr) [reply xml/obs_basic_workflow.xml]: Ok, done

File: xml/obs_basic_workflow.xml (hunk 12 above)
[the remark this answers is not in the document]
Comment (sknorr) [reply xml/obs_basic_workflow.xml]: That does not work imo  --
1. it does not communicate that the real source files are now removed from the checkout
2. "contains" communicates a state but it does not communicate a change

**Pull request closed without merging**